Makefile

```diff
@@ -241,42 +241,66 @@ $(BUILT)/add-on/install.rdf: add-on/install.rdf.in \
+shared_files=$(shared_js_files) $(shared_jsx_files:.jsx=.js) $(shared_test_files)
+built_add_on_shared_js_files=$(addprefix $(BUILT)/add-on/chrome/content/, $(shared_files))
+built_standalone_shared_js_files=$(addprefix $(BUILT)/standalone/content/, $(shared_files))
+built_ui_shared_js_files=$(addprefix $(BUILT)/ui/, $(shared_files))
 
 #
 # The following sections are for the non-shared assets:
 #
 
 # ui-showcase
 ui_jsx_files=$(wildcard ui/*.jsx)
-built_ui_js_files=$(ui_jsx_files:%.jsx=$(BUILT)/%.js)
+ui_js_files=$(wildcard ui/*.js)
+ui_files=$(ui_jsx_files:%.jsx=%.js) $(ui_js_files)
+built_ui_js_files= $(addprefix $(BUILT)/, $(ui_files))
 
 # standalone
 standalone_jsx_files=$(wildcard standalone/content/js/*.jsx)
-built_standalone_js_files=$(standalone_jsx_files:%.jsx=$(BUILT)/%.js)
+standalone_js_files=$(wildcard standalone/content/js/*.js)
+standalone_test_files=$(wildcard standalone/test/*.js)
+standalone_misc_files=$(wildcard standalone/*.js) \
+	$(wildcard standalone/content/*.js)
+built_standalone_js_files= \
+	$(standalone_jsx_files:%.jsx=$(BUILT)/%.js) \
+	$(standalone_js_files:%.js=$(BUILT)/%.js) \
+	$(standalone_test_files:%.js=$(BUILT)/%.js) \
+	$(standalone_misc_files:%.js=$(BUILT)/%.js)
 
 standalone_l10n_files=$(wildcard locale/*/standalone.properties)
-built_standalone_l10n_files=$(patsubst locale/%/standalone.properties, \
-	$(BUILT)/standalone/content/l10n/%/loop.properties, \
-	$(standalone_l10n_files))
+built_standalone_l10n_files= \
+	$(standalone_l10n_files:locale/%/standalone.properties=$(BUILT)/standalone/content/l10n/%/loop.properties)
 
 $(BUILT)/standalone/content/l10n/%/loop.properties: locale/%/standalone.properties locale/%/shared.properties
 	@mkdir -p $(@D)
 	cat $^ > $@
 
 # add-on
+add_on_js_files=$(wildcard add-on/panels/js/*.js)
 add_on_jsx_files=$(wildcard add-on/panels/js/*.jsx)
-built_add_on_js_files=$(patsubst add-on/panels/js/%.jsx, \
-	 $(BUILT)/add-on/chrome/content/panels/js/%.js, \
-	 $(add_on_jsx_files))
+add_on_module_js_files=$(wildcard add-on/chrome/modules/*.js)
+add_on_module_jsm_files=$(wildcard add-on/chrome/modules/*.jsm)
+add_on_test_files=$(wildcard add-on/panels/test/*.js)
+add_on_files=$(add_on_js_files) \
+	$(add_on_jsx_files:.jsx=.js) \
+	$(add_on_module_js_files) \
+	$(add_on_module_jsm_files) \
+	$(add_on_test_files)
+built_add_on_js_files=$(addprefix $(BUILT)/, \
+	$(subst chrome/,chrome/content/, $(subst panels/,chrome/panels/, $(add_on_files))))
 
 add_on_vendor_jsx_files=$(wildcard add-on/panels/vendor/*.jsx)
+add_on_vendor_js_files=$(wildcard add-on/panels/vendor/*.js)
 built_add_on_vendor_js_files=$(patsubst add-on/panels/vendor/%.jsx, \
-	 $(BUILT)/add-on/chrome/content/panels/vendor/%.js, \
-	 $(add_on_vendor_jsx_files))
+	$(BUILT)/add-on/chrome/content/panels/vendor/%.js, \
+	$(add_on_vendor_jsx_files)) \
+	$(patsubst add-on/panels/vendor/%.js, \
+	$(BUILT)/add-on/chrome/content/panels/vendor/%.js, \
+	$(add_on_vendor_js_files))
```

Comment:
These are still using `patsubst`s split across multiple lines oddly indented. Might as well just remove `built_add_on_vendor_js_files` and include them as part of `add_on_files`.